### PR TITLE
feat(real-ip): implement the first version

### PR DESCRIPTION
**Pull request by spacewander**
Signed-off-by: spacewander <spacewanderlzx@gmail.com>

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [x] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: docs/en/latest/plugins/real-ip.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+It works like Nginx's `ngx_http_realip_module`, but is more flexible.
+
+This plugin requires APISIX to run on [APISIX-OpenResty](../how-to-build.md#step-6-build-openresty-for-apache-apisix).
+
+## Attributes
+
+| Name      | Type          | Requirement | Default    | Valid                                                                    | Description                                                                                                                                         |
+| --------- | ------------- | ----------- | ---------- | ------------------------------------------------------------------------ | --------------------------------------------------------------------------------------------------------------------------------------------------- |
+| source      | string        | required    |            | Any Nginx variable like `arg_realip` or `http_x_forwarded_for`| dynamically set the client's IP and port in APISIX's view, according to the value of variable. If the value doesn't contain a port, the client's port won't be changed. |
```
Comment (tokers) [inline docs/en/latest/plugins/real-ip.md:44]: We may also hint to the users that if the address format is not correct, the `real-ip` plugin will just let it go and don't change the client address.

**Review by tzssangglass — APPROVED** [review]

File: docs/en/latest/plugins/real-ip.md (hunk 1 above)
Comment (spacewander) [reply docs/en/latest/plugins/real-ip.md:44]: Added in the new commit

Comment (spacewander) [reply docs/en/latest/plugins/real-ip.md:44]: Added in the new commit

File: docs/en/latest/plugins/real-ip.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+- [**Test Plugin**](#test-plugin)
+- [**Disable Plugin**](#disable-plugin)
+
+## Name
+
+The `real-ip` plugin dynamically changes the client's IP and port seen by APISIX.
+
+It works like Nginx's `ngx_http_realip_module`, but is more flexible.
+
+This plugin requires APISIX to run on [APISIX-OpenResty](../how-to-build.md#step-6-build-openresty-for-apache-apisix).
```
Comment (membphis) [inline docs/en/latest/plugins/real-ip.md:38]: add a `note` is better

**Review by membphis — APPROVED** [review]

**Review by tokers — APPROVED** [review]

**Pull request merged**